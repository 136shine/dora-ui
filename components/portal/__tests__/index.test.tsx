import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Portal from '../index';

describe('<Portal />', () => {
  test('should append node to <body /> by default', () => {
    const { baseElement } = render(
      <Portal>
        <div>hello,portal</div>
      </Portal>
    );
    expect(baseElement).toMatchSnapshot();
  });

  test('should remove the default container when Portal unmounted', () => {
    const { baseElement, unmount } = render(
      <Portal>
        <div>hello,portal</div>
      </Portal>
    );
    expect(baseElement).toMatchSnapshot();
    unmount();
    expect(baseElement).toMatchSnapshot();
  });

  test('should append node to custom container by props "node"', () => {
    const div = document.createElement('div');
    div.className = 'custom-container';
    document.body.append(div);
    const { baseElement, unmount } = render(
      <Portal node={div}>
        <div>custom container</div>
      </Portal>
    );
    expect(baseElement).toMatchSnapshot();
    unmount();
    expect(baseElement).toMatchSnapshot();
    // custom container will not be removed
    document.body.removeChild(div);
  });
});
