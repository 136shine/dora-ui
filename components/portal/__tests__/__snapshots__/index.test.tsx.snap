// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Portal /> should append node to <body /> by default 1`] = `
<body>
  <div />
  <div>
    <div>
      hello,portal
    </div>
  </div>
</body>
`;

exports[`<Portal /> should append node to custom container by props "node" 1`] = `
<body>
  <div
    class="custom-container"
  >
    <div>
      custom container
    </div>
  </div>
  <div />
</body>
`;

exports[`<Portal /> should append node to custom container by props "node" 2`] = `
<body>
  <div
    class="custom-container"
  />
  <div />
</body>
`;

exports[`<Portal /> should remove the default container when Portal unmounted 1`] = `
<body>
  <div />
  <div>
    <div>
      hello,portal
    </div>
  </div>
</body>
`;

exports[`<Portal /> should remove the default container when Portal unmounted 2`] = `
<body>
  <div />
</body>
`;
